Sources/ApodiniDeploy/ApodiniDeployInterfaceExporter.swift

```diff
@@ -0,0 +1,250 @@
+//
+//  ApodiniDeployInterfaceExporter.swift
+//  
+//
+//  Created by Lukas Kollmer on 2021-01-14.
+//
+
+import Foundation
+import Apodini
+import ApodiniUtils
+import ApodiniVaporSupport
+import ApodiniDeployBuildSupport
+import ApodiniDeployRuntimeSupport
+@_implementationOnly import Vapor
+@_implementationOnly import AssociatedTypeRequirementsVisitor
+
+
+
+extension AnyEndpointParameter {
+    var lk_stableIdentity: String {
+        switch self.parameterType {
+        case .content:
+            return "c:\(self.name)"
+        case .lightweight:
+            return "l:\(self.name)"
+        case .path:
+            return "p:\(self.name)"
+        }
+    }
+}
+
+
+
+
+struct ApodiniDeployError: Swift.Error {
+    let message: String
+}
+
+
+/// A custom internal interface exporter, which:
+/// a) compiles a list of all handlers (via their `Endpoint` objects). These are used to determine the target endpoint when manually invoking a handler.
+/// b) is responsible for handling parameter retrieval when manually invoking handlers.
+/// c) exports an additional endpoint used to manually invoke a handler remotely over the network.
+public class ApodiniDeployInterfaceExporter: InterfaceExporter {
+    struct ApplicationStorageKey: Apodini.StorageKey {
+        typealias Value = ApodiniDeployInterfaceExporter
+    }
+    
+    struct CollectedEndpointInfo {
+        let handlerType: HandlerTypeIdentifier
+        let endpoint: AnyEndpoint
+        let deploymentOptions: DeploymentOptions
+    }
+    
+    
+    let app: Apodini.Application
+    
+    var collectedEndpoints: [CollectedEndpointInfo] = []
+    var explicitlyCreatedDeploymentGroups: [DeploymentGroup.ID: Set<AnyHandlerIdentifier>] = [:]
+    
+    var deploymentProviderRuntime: DeploymentProviderRuntimeSupport?
+    
+    
+    public required init(_ app: Apodini.Application) {
+        self.app = app
+        app.storage.set(ApplicationStorageKey.self, to: self)
+    }
+    
+    
+    public func export<H: Handler>(_ endpoint: Endpoint<H>) {
+        if let groupId = endpoint.context.get(valueFor: DSLSpecifiedDeploymentGroupIdContextKey.self) {
+            if explicitlyCreatedDeploymentGroups[groupId] == nil {
+                explicitlyCreatedDeploymentGroups[groupId] = []
+            }
+            explicitlyCreatedDeploymentGroups[groupId]!.insert(endpoint.identifier)
+        }
+        collectedEndpoints.append(CollectedEndpointInfo(
+            handlerType: HandlerTypeIdentifier(H.self),
+            endpoint: endpoint,
+            deploymentOptions: CollectedOptions(reducing: [
+                endpoint.handler.getDeploymentOptions(),
+                endpoint.context.get(valueFor: HandlerDeploymentOptionsSyntaxNodeContextKey.self)
+            ].flatMap { $0.compactMap { $0.resolve(against: endpoint.handler) } })
+        ))
+        app.vapor.app.add(Vapor.Route(
```

Comment:
Correct, these routes currently would not be accessible via the RemoteHandlerInvocation API. I figured it wouldn't hurt to add them anyway, in case there might be other reasons for which we'd want to manually invoke a specific handler in a specific instance (eg things like the observed objects, etc).

Reply:
I see. I think we should somehow make this configurable. It might be useful for debug purposes or to try out something but I think it would be an unexpected behavior if a production system would expose more routes then expected (even though it would just be duplicates as the `Handler` might already be exposed by an other interface exporter)

Reply:
Yes that's a good point, i'll limit it to `InvocableHandler`s for the time being, and, if at some point in the future there is an API for defining exporter configurations (the way i'm imagining this it would be part of the stackable exporters, since the internal invocation API would probably be used by some other exporter or part of Apodini) we can bring it back via that.